src/beast/beast/crypto/impl/ripemd_context.h

```diff
@@ -0,0 +1,441 @@
+
+    unsigned int tot_len;
+    unsigned int len;
+    std::uint32_t block[64];
+    std::uint32_t h[5];
+};
+
+// ROL(x, n) cyclically rotates x over n bits to the left
+// x must be of an unsigned 32 bits type and 0 <= n < 32.
+#define BEAST_RIPEMD_ROL(x, n)  (((x) << (n)) | ((x) >> (32-(n))))
+
+// the five basic functions F(), G() and H()
+#define BEAST_RIPEMD_F(x, y, z)  ((x) ^  (y) ^ (z)) 
+#define BEAST_RIPEMD_G(x, y, z) (((x) &  (y)) | (~(x) & (z))) 
+#define BEAST_RIPEMD_H(x, y, z) (((x) | ~(y)) ^ (z))
+#define BEAST_RIPEMD_I(x, y, z) (((x) &  (z)) | ((y) & ~(z))) 
+#define BEAST_RIPEMD_J(x, y, z)  ((x) ^  ((y) | ~(z)))
+  
+// the ten basic operations FF() through III()
+#define BEAST_RIPEMD_FF(a, b, c, d, e, x, s) {      \
+      (a) += BEAST_RIPEMD_F((b), (c), (d)) + (x);   \
+      (a) = BEAST_RIPEMD_ROL((a), (s)) + (e);       \
+      (c) = BEAST_RIPEMD_ROL((c), 10); }
+#define BEAST_RIPEMD_GG(a, b, c, d, e, x, s) {      \
+      (a) += BEAST_RIPEMD_G((b), (c), (d)) + (x) + 0x5a827999UL; \
+      (a) = BEAST_RIPEMD_ROL((a), (s)) + (e);       \
+      (c) = BEAST_RIPEMD_ROL((c), 10); }
+#define BEAST_RIPEMD_HH(a, b, c, d, e, x, s) {      \
+      (a) += BEAST_RIPEMD_H((b), (c), (d)) + (x) + 0x6ed9eba1UL;\
+      (a) = BEAST_RIPEMD_ROL((a), (s)) + (e);       \
+      (c) = BEAST_RIPEMD_ROL((c), 10); }
+#define BEAST_RIPEMD_II(a, b, c, d, e, x, s) {      \
+      (a) += BEAST_RIPEMD_I((b), (c), (d)) + (x) + 0x8f1bbcdcUL;\
+      (a) = BEAST_RIPEMD_ROL((a), (s)) + (e);       \
+      (c) = BEAST_RIPEMD_ROL((c), 10); }
+#define BEAST_RIPEMD_JJ(a, b, c, d, e, x, s) {      \
+      (a) += BEAST_RIPEMD_J((b), (c), (d)) + (x) + 0xa953fd4eUL;\
+      (a) = BEAST_RIPEMD_ROL((a), (s)) + (e);       \
+      (c) = BEAST_RIPEMD_ROL((c), 10); }
+#define BEAST_RIPEMD_FFF(a, b, c, d, e, x, s) {     \
+      (a) += BEAST_RIPEMD_F((b), (c), (d)) + (x);   \
+      (a) = BEAST_RIPEMD_ROL((a), (s)) + (e);       \
+      (c) = BEAST_RIPEMD_ROL((c), 10); }
+#define BEAST_RIPEMD_GGG(a, b, c, d, e, x, s) {     \
+      (a) += BEAST_RIPEMD_G((b), (c), (d)) + (x) + 0x7a6d76e9UL;\
+      (a) = BEAST_RIPEMD_ROL((a), (s)) + (e);       \
+      (c) = BEAST_RIPEMD_ROL((c), 10); }
+#define BEAST_RIPEMD_HHH(a, b, c, d, e, x, s) {     \
+      (a) += BEAST_RIPEMD_H((b), (c), (d)) + (x) + 0x6d703ef3UL;\
+      (a) = BEAST_RIPEMD_ROL((a), (s)) + (e);       \
+      (c) = BEAST_RIPEMD_ROL((c), 10); }
+#define BEAST_RIPEMD_III(a, b, c, d, e, x, s) {     \
+      (a) += BEAST_RIPEMD_I((b), (c), (d)) + (x) + 0x5c4dd124UL;\
+      (a) = BEAST_RIPEMD_ROL((a), (s)) + (e);       \
+      (c) = BEAST_RIPEMD_ROL((c), 10);  }
+#define BEAST_RIPEMD_JJJ(a, b, c, d, e, x, s) {     \
+      (a) += BEAST_RIPEMD_J((b), (c), (d)) + (x) + 0x50a28be6UL;\
+      (a) = BEAST_RIPEMD_ROL((a), (s)) + (e);       \
+      (c) = BEAST_RIPEMD_ROL((c), 10); }
+
+template <class = void>
+void ripemd_load (std::array<std::uint32_t, 16>& X,
+    void const* block)
+{
+    std::uint8_t const* p =
```

Comment:
Good use for `auto` since you are just repeating the type you have a few lines below.